lib/should.js

```
@@ -106,8 +106,35 @@ var Assertion = should.Assertion = function Assertion(obj) {
   this.obj = obj;
 };
 
+
 var hasOwnProperty = Object.prototype.hasOwnProperty;
 
+var blockify = function (block) {
+  if (typeof block === 'function' && /function\s+\w*\((\w+)\)[\s\S]*\w\.should\.\w+/.test(block.toString())) {
+    return function (obj) {
+      try {
```

I think you do not need to catch all exceptions. E.g. if it is AssertionError it can be usefull to reuse it, other exceptions should go thought.